datafusion/spark/src/function/math/abs.rs

```diff
@@ -0,0 +1,432 @@
+        Ok(ColumnarValue::Scalar(ScalarValue::$SCALAR_TYPE(Some(
+            result,
+        ))))
+    }};
+    ($INPUT:ident, $PRECISION:expr, $SCALE:expr, $SCALAR_TYPE:ident) => {{
+        let result = $INPUT.wrapping_abs();
+        Ok(ColumnarValue::Scalar(ScalarValue::$SCALAR_TYPE(
+            Some(result),
+            $PRECISION,
+            $SCALE,
+        )))
+    }};
+}
+
+pub fn spark_abs(args: &[ColumnarValue]) -> Result<ColumnarValue, DataFusionError> {
+    if args.len() != 1 {
+        return internal_err!("abs takes exactly 1 argument, but got: {}", args.len());
+    }
+
+    match &args[0] {
+        ColumnarValue::Array(array) => match array.data_type() {
+            DataType::Null
+            | DataType::UInt8
+            | DataType::UInt16
+            | DataType::UInt32
+            | DataType::UInt64 => Ok(args[0].clone()),
+            DataType::Int8 => {
+                let abs_fun = make_wrapping_abs_function!(Int8Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Int16 => {
+                let abs_fun = make_wrapping_abs_function!(Int16Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Int32 => {
+                let abs_fun = make_wrapping_abs_function!(Int32Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Int64 => {
+                let abs_fun = make_wrapping_abs_function!(Int64Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Float32 => {
+                let abs_fun = make_abs_function!(Float32Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Float64 => {
+                let abs_fun = make_abs_function!(Float64Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Decimal128(_, _) => {
+                let abs_fun = make_wrapping_abs_function!(Decimal128Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            DataType::Decimal256(_, _) => {
+                let abs_fun = make_wrapping_abs_function!(Decimal256Array);
+                abs_fun(array).map(ColumnarValue::Array)
+            }
+            dt => internal_err!("Not supported datatype for Spark ABS: {dt}"),
+        },
+        ColumnarValue::Scalar(sv) => match sv {
+            ScalarValue::Null
+            | ScalarValue::UInt8(_)
+            | ScalarValue::UInt16(_)
+            | ScalarValue::UInt32(_)
+            | ScalarValue::UInt64(_) => Ok(args[0].clone()),
+            ScalarValue::Int8(a) => match a {
+                None => Ok(args[0].clone()),
+                Some(v) => scalar_compute_op!(v, Int8),
+            },
+            ScalarValue::Int16(a) => match a {
+                None => Ok(args[0].clone()),
+                Some(v) => scalar_compute_op!(v, Int16),
+            },
```

Comment:
```suggestion
            | ScalarValue::UInt64(_) => Ok(args[0].clone()),
            sv if sv.is_null() => Ok(args[0].clone()),
            ScalarValue::Int8(Some(v)) => scalar_compute_op!(v, Int8),
            ScalarValue::Int16(Some(v)) => scalar_compute_op!(v, Int16),
```

Slight more compact way of doing this